blendergltf.py

```diff
@@ -874,6 +874,32 @@ def export_gltf(scene_delta, settings={}):
     else:
         mesh_list = scene_delta.get('meshes', [])
 
+    # Use this to store objects that we need to transform back from their
+    # converted axis space. There is no need to copy meshes that don't have any
+    # modifiers applied but that means we have to be careful in modifying them.
+    user_objects = []
+
+    global_matrix = settings.get('global_matrix', None)
+    if global_matrix != None:
+        # If a global matrix was provided apply it to all objects.
+        for obj in object_list:
+            # Find the mesh associated with this object
+            if obj.name in mod_meshes:
+                # Use the modified mesh copy (completely safe to modify).
+                mesh = mod_meshes[obj.name]
+            else:
+                # We have to remember to undo the transformation because this is
+                # a user mesh (not one we created).
+                mesh = obj.data
+                user_objects.append(obj)
+
+            inv_world_mat = obj.matrix_world.inverted()
+
+            # Transform the mesh from model space to blender world space, then
+            # converted-axis world space and finally back to a converted-axis
+            # model space, suitable for copying.
+            mesh.transform(inv_world_mat * global_matrix * obj.matrix_world)
```

Comment:
Just tested it and found the mesh *is* transformed multiple times and this results in the objects not changing orientation but moving some direction from the origin. It didn't seem to affect the undo stack because when I went to undo after the export, it undoed past my edit mode modification I made *before* exporting. So in effect it actually got of rid of both changes (the edit mode change and the blendergltf change), although I understand this to mean the undo stack *was not* affected.

Looking at the .obj exporter it looks like at export each object is copied (with modifiers applied or not) and then transformed after the copy. That being said, we have more options to express the axis conversion / world transform like you said whereas the .obj just outputs world coordinates of whatever objects are exported.

In my original code (which applied modifiers and made a new mesh for every object), the names did get screwed up, but this could probably be fixed by attaching the original name to the new mesh using another map or using a tuple, etc.